DSharpPlus.CommandsNext/Converters/EntityConverters.cs

```diff
@@ -62,16 +62,15 @@ async Task<Optional<DiscordUser>> IArgumentConverter<DiscordUser>.ConvertAsync(s
             }
 
             var cs = ctx.Config.CaseSensitive;
-            if (!cs)
-                value = value.ToLowerInvariant();
 
             var di = value.IndexOf('#');
             var un = di != -1 ? value.Substring(0, di) : value;
             var dv = di != -1 ? value.Substring(di + 1) : null;
 
             var us = ctx.Client.Guilds.Values
-                .SelectMany(xkvp => xkvp.Members.Values)
-                .Where(xm => (cs ? xm.Username : xm.Username.ToLowerInvariant()) == un && ((dv != null && xm.Discriminator == dv) || dv == null));
+                .SelectMany(xkvp => xkvp.Members.Values).Where(xm =>
+                    xm.Username.Equals(un, cs ? StringComparison.InvariantCulture : StringComparison.InvariantCultureIgnoreCase) &&
+                    ((dv != null && xm.Discriminator == dv) || dv == null));
```

Comment:
In case of `DiscordUser` and `DiscordMember` converters i will do tests for `DiscordUser` only, unless requested otherwise, since code there is pretty much the same.
We just need to call command where `DiscordUser` is parameter. And to make sure not to send mention object but to convert user object from `username#discriminator` format
`d# sudo testingBot#2657 avatar`, version where bot name is written in different case
`d# sudo testingbot#2657 avatar`
![image](https://user-images.githubusercontent.com/50947930/152697567-82da943c-5929-4744-8814-4673644498a0.png)